src/profile-logic/marker-data.js

```diff
@@ -14,82 +14,151 @@ import type {
  */
 export function deriveJankMarkers(
   samples: SamplesTable,
   thresholdInMs: number,
+  eventDelayTable: EventDelayTable | null,
   otherCategoryIndex: IndexIntoCategoryList
 ): Marker[] {
-  const addMarker = () =>
+  const addMarker = (start, duration) =>
     jankInstances.push({
-      start: lastTimestamp - lastResponsiveness,
-      dur: lastResponsiveness,
-      title: `${lastResponsiveness.toFixed(2)}ms event processing delay`,
+      start: start,
+      dur: duration,
+      title: `${duration.toFixed(2)}ms event processing delay`,
       name: 'Jank',
       category: otherCategoryIndex,
       data: null,
     });
 
-  let lastResponsiveness: number = 0;
-  let lastTimestamp: number = 0;
   const jankInstances = [];
-  for (let i = 0; i < samples.length; i++) {
-    let currentResponsiveness;
-    if (samples.eventDelay) {
-      currentResponsiveness = samples.eventDelay[i];
-    } else if (samples.responsiveness) {
-      currentResponsiveness = samples.responsiveness[i];
+  // As mentioned in the comment above, we have to support two values types.
+  // They are slightly different from each other.
+  if (eventDelayTable !== null) {
+    // This is a new profile with eventDelays and we should take a look at the
+    // processed event delay values to be able to create Jank markers.
+    const { eventDelays } = eventDelayTable;
+
+    let lastEventDelay: Milliseconds = 0;
+    for (let i = 0; i < samples.length; i++) {
+      const currentEventDelay = eventDelays[i];
+      if (currentEventDelay === null || currentEventDelay === undefined) {
+        // Ignore anything that's not numeric. This can happen if there is no responsiveness
+        // information, or if the sampler failed to collect a responsiveness value. This
+        // can happen intermittently.
+        //
+        // See Bug 1506226.
+        continue;
+      }
+      if (currentEventDelay > lastEventDelay) {
+        if (currentEventDelay >= thresholdInMs) {
+          addMarker(samples.time[i], currentEventDelay);
+        }
+      }
+      lastEventDelay = currentEventDelay;
     }
-
-    if (currentResponsiveness === null || currentResponsiveness === undefined) {
-      // Ignore anything that's not numeric. This can happen if there is no responsiveness
-      // information, or if the sampler failed to collect a responsiveness value. This
-      // can happen intermittently.
-      //
-      // See Bug 1506226.
-      continue;
+    // Check the last event delay to see if the threshold is reached.
+    if (lastEventDelay >= thresholdInMs) {
+      addMarker(
+        samples.time[samples.length - 1],
+        eventDelays[samples.length - 1]
+      );
     }
-    if (currentResponsiveness < lastResponsiveness) {
-      if (lastResponsiveness >= thresholdInMs) {
-        addMarker();
+  } else {
+    // This is an old profile which still has `responsiveness` values. We are
+    // keeping the old algorithm here to still support the older profiles.
```

Comment:
It would be really good if we can implement an upgrader. This can be in a follow-up.